pkgs/servers/sql/postgresql/libpq.nix

```diff
@@ -0,0 +1,70 @@
+{ stdenv
+, lib
+, openssl
+, zlib
+, libkrb5
+, postgresql
+, perl
+, bison
+, flex
+, tzdata
+, gssSupport ? with stdenv.hostPlatform; !isWindows && !isStatic
+}:
+
+stdenv.mkDerivation {
+  pname = "libpq";
+  version = postgresql.version;
+  src = postgresql.src;
+
+  configureFlags = [
+    "--with-openssl"
+    "--without-readline"
+    "--with-system-tzdata=${tzdata}/share/zoneinfo"
+  ]
+  ++ lib.optionals gssSupport [ "--with-gssapi" ];
+
+  nativeBuildInputs = [ perl bison flex tzdata ];
+  buildInputs = [ openssl zlib ]
+    ++ lib.optional gssSupport libkrb5;
+
+  buildPhase = ''
+    runHook preBuild
+
+    make -C src/bin/pg_config
+    make -C src/common
+    make -C src/include
+    make -C src/interfaces/libpq
+    make -C src/port
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    make -C src/bin/pg_config install
+    make -C src/common install
+    make -C src/include install
+    make -C src/interfaces/libpq install
+    make -C src/port install
+
+    mkdir -p $dev/{bin,lib}
+    mv $out/bin/pg_config $dev/bin/pg_config
+    mv $out/lib/*.a $dev/lib/
```

Comment:
Possibly move to `static` output?

Reply:
Anecdotally I've had it accidentally pick up static outputs instead of dynamic ones at some point. But I'm struggling to reproduce it. That would be one case for having a separate `static` output. We could optionally include it in `dev` if `isStatic` is set.

Reply:
I tried build this simple program with this libpq:
```c
extern char PQconnectdb();
int
main (void)
{
PQconnectdb()
  ;
  return 0;
}
```
(build with `-lpq`)
and it gives me a huge number of undefined references. It turns out the the static library is used, which needs all of its dependencies.

Do we really need the static library to exist at all?

Reply:
> Do we really need the static library to exist at all?

I don't think it has to be a blocker. Though one of the reasons people were asking for a separate `libpq` package was that `postgresql.lib` was difficult to build statically.

Reply:
Fixed. It also works with static builds now using `libpq.static`. I tested both.